lib/common/components/passwordInputDialog.dart

```diff
@@ -83,6 +84,14 @@ class _PasswordInputDialog extends State<PasswordInputDialog> {
         ),
       ),
       actions: <Widget>[
+        CupertinoButton(
+          child: Text(dic['reset']),
+          onPressed: () {
+            print("onReset is:");
+            print(widget.onReset);
+            widget.onReset != null ? widget.onReset() : Navigator.of(context).pop();
+          },
+        ),
```

Comment:
This should change, I believe. Do I understand correctly that every password input dialog has now the reset button, even if the `onReset` function is not set? Also remove those debug prints, please.

I see that you copied the logic from the cancel button, but I think it does not make sense here.

```suggestion
        widget.onReset != null :
        CupertinoButton(
          child: Text(dic['reset']),
          onPressed: () {
            widget.onReset()
          },
        ) : Container();
```

Reply:
> This should change, I believe. Do I understand correctly that every password input dialog has now the reset button, even if the `onReset` function is not set? Also remove those debug prints, please.
> 
> I see that you copied the logic from the cancel button, but I think it does not make sense here.

Yes, this is in every password input dialog. And yes, even if onReset is not set, it will be shown, and it would trigger a Navigator.of(context).pop() if not defined. 
I guess its just a safety net for now to not crash, if someone didn't define the onReset, but a better solution would be, if it would be possible to only show the reset button if defined. I'm not sure this is actually possible. 
Another solution would be to create a total new widget with the ResetButton such that we can differentiate when to use the specific dialog

Reply:
Oh, I see now what you actually suggested! very nice ^^ I will implement this change